core/include/scipp/core/transform.h

```diff
@@ -235,10 +235,51 @@ struct is_eigen_type<sparse_container<Eigen::Matrix<T, Rows, Cols>>>
 template <class T>
 inline constexpr bool is_eigen_type_v = is_eigen_type<T>::value;
 
+namespace transform_detail {
+template <class T> struct is_sparse : std::false_type {};
+template <class T> struct is_sparse<sparse_container<T>> : std::true_type {};
+template <class T>
+struct is_sparse<ValuesAndVariances<sparse_container<T>>> : std::true_type {};
+template <class T>
+struct is_sparse<ValuesAndVariances<const sparse_container<T>>>
+    : std::true_type {};
+template <class T> inline constexpr bool is_sparse_v = is_sparse<T>::value;
+} // namespace transform_detail
+
+template <class T> auto check_and_get_size(const T &a) {
+  return scipp::size(a);
```

Comment:
Technically for std containers as `T`, this is this could be performing a narrowing operation, though triggering that condition is not practice, and therefore nothing to do here.

Reply:
I am using signed indices every in scipp. It is 64 bit and I think this is essentially just what `std::ssize` (C++20) does, see https://en.cppreference.com/w/cpp/iterator/size.

Reply:
Indeed. I wasn't aware of `ssize`.